backend/store/v2/etcdstore/watcher.go

```diff
@@ -0,0 +1,149 @@
+package etcdstore
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/prometheus/client_golang/prometheus"
+	storev2 "github.com/sensu/sensu-go/backend/store/v2"
+	"github.com/sensu/sensu-go/backend/store/v2/wrap"
+	"go.etcd.io/etcd/api/v3/mvccpb"
+	clientv3 "go.etcd.io/etcd/client/v3"
+)
+
+const (
+	WatchEventsCounterVec = "sensu_go_watch_events"
+
+	WatchEventsLabelStatus       = "status"
+	WatchEventsLabelResourceType = "resource"
+	WatchEventsLabelNamespace    = "namespace"
+
+	WatchEventsStatusHandled = "handled"
+	WatchEventsStatusDropped = "dropped"
+)
+
+var (
+	WatchEventsProcessed = prometheus.NewCounterVec(
```

Comment:
It's probably in our best interest to expose this metric from the postgres store implementation too. Thoughts on moving this to the backend/store/v2 package?

Reply:
Agreed!